### PR TITLE
Change: [CI] Use a custom name for matrix runs

### PR DESCRIPTION
**Pull request by TrueBrain**
## Motivation / Problem

By default, GitHub adds all arguments of the matrix between (). This is fine sometimes, but in other times it becomes a very lengthy line.

## Description

With this commit, we decide what is between those (), making it a lot more readable.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, gs_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

File: .github/workflows/ci-build.yml [hunk 1]
```diff
@@ -69,24 +69,29 @@ jobs:
...
 
   linux:
-    name: Linux
-
     strategy:
       fail-fast: false
       matrix:
         include:
-        - compiler: clang
+        - name: Ckang
```
Comment (glx22) [inline .github/workflows/ci-build.yml]: oups

File: .github/workflows/ci-build.yml [hunk 2]
```diff
@@ -349,6 +352,8 @@ jobs:
           - msystem: MINGW32
             arch: i686
 
+    name: MingW (${{ matrix.arch }})
```
Comment (LordAro) [inline .github/workflows/ci-build.yml]:
```suggestion
    name: MinGW (${{ matrix.arch }})
```

**Comment by rubidium42** [comment]
Is the configuration of which checks are required and which aren't required in the repository, or is that something separately? I can't find it, so I'm assuming it's the latter.

If it's the former, then that needs to be updated as it is awaiting a lot of required "checks" that will never be run. 

If it's the latter, then someone needs to change the "required checks configuration" and all PRs after this gets merged need to be rebased to pass these changed checks, right?

**Comment by TrueBrain** [comment]
After this PR is approved, I need to change some GitHub settings so the CI required labels change; and yes, after that all PRs need to rebase :)

**Review by rubidium42 — APPROVED** [review]

**Pull request merged**